### PR TITLE
1651 - Refactor job group filter to magic numbers

### PR DESCRIPTION
**Pull request by AchyutRatkanthiwar**
## Description
Trello ticket [#1651](https://trello.com/c/WWsAecYp/1651-jr-filtering-3-refactor-job-group-filter-to-magic-numbers)

refactored the `filter_job_role_group_outliers` function within clean step of estimates by job role pipeline to use magic numbers as outlier bounds. These bounds are calculated as part of another [ticket](https://trello.com/c/sE8DkZxz/1650-jr-filtering-2-recalculating-bounds) and are saved [here](https://skillsforcare.atlassian.net/wiki/spaces/DE/pages/1875804161/Job+Role+Filtering)

## Testing
- [x] Unit tests passing
- [x] Successful [Step Function run](https://eu-west-2.console.aws.amazon.com/states/home?region=eu-west-2#/v2/executions/details/arn:aws:states:eu-west-2:856699698263:execution:1651-fr-filter-3-Ind-CQC-Filled-Post-Estimates-By-Role:7b4d72d7-91eb-4d0f-99b8-1ffc8295c202)
- [x] Outputs checked in Athena

[Add any additional testing information here - add screenshot if relevant]

## Checklist (delete if not relevant)
- [x] Unit tests added/amended
- [x] Docstrings added/updated
- [x] Updated CHANGELOG
- [x] Code reviewed by AI
- [x] Moved Trello ticket to PR column

## Reviewer checklist
- [ ] PR solves the Trello ticket requirements
- [ ] Outputs appear reasonable and understandable
- [ ] The overall approach is appropriate and not over-engineered
- [ ] No obvious scalability, performance or interdependency concerns
- [ ] Tests appropriately cover the main behaviour/edge cases
- [ ] Naming and structure are broadly understandable and consistent
- [ ] Docstrings are sufficient for future maintenance
- [ ] Docstrings cover non-obvious behaviour

### CODE REVIEW

**Comment by MarionJHolloway** [comment]
Your step function failed. I will review the rest of the PR, but you might want to look into that whilst I do :)

File: projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/fargate/utils/clean_utils.py [hunk 1]
```diff
@@ -137,40 +138,57 @@ def filter_job_role_group_outliers(
...
     job_group_percentage_expr: pl.Expr
     evaluation_expr: pl.Expr
-    upper_bounds_expr: pl.Expr
-    lower_bounds_expr: pl.Expr
 
-    def __init__(self, upper_percentile_bound: float, lower_percentile_bound: float):
+    def __init__(self):
+        self.job_role_group_bounds: dict[str, dict[str, float]] = {
+            PrimaryServiceType.care_home_only: {
+                f"{JobGroupLabels.direct_care}_upper": 0.985761,
```
Comment (MarionJHolloway) [inline projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/fargate/utils/clean_utils.py]: Use the suffixes here as well?

File: projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/tests/fargate/utils/test_clean_utils.py [hunk 2]
```diff
@@ -100,9 +101,33 @@ def test_filter_when_job_group_ratio_outside_percentile_bounds(self, case):
...
 class TestFilterJobRoleGroupExpressions:
-    test_upper_bound = 0.8
-    test_lower_bound = 0.2
-    TestExprs = job.FilterJobRoleGroupExpressions(test_upper_bound, test_lower_bound)
+    TestExprs = job.FilterJobRoleGroupExpressions()
+
+    def test_job_role_group_bounds_dict(self):
+        expected_bounds = {
+            PrimaryServiceType.care_home_only: {
+                f"{JobGroupLabels.direct_care}_upper": 0.985761,
```
Comment (MarionJHolloway) [inline projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/tests/fargate/utils/test_clean_utils.py]: Use suffixes here too?

**Review by MarionJHolloway — COMMENTED** [review]
A few small comments. Were you able to reduce the test datasets at all after this change?

File: projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/fargate/utils/clean_utils.py (hunk 1 above)
Comment (AchyutRatkanthiwar) [reply projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/fargate/utils/clean_utils.py]: Done

File: projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/tests/fargate/utils/test_clean_utils.py (hunk 2 above)
Comment (AchyutRatkanthiwar) [reply projects/_03_independent_cqc/_07_estimate_filled_posts_by_job_role/tests/fargate/utils/test_clean_utils.py]: Done

**Comment by AchyutRatkanthiwar** [comment]
> A few small comments. Were you able to reduce the test datasets at all after this change?

Yes, the test data is much smaller now.

**Comment by AchyutRatkanthiwar** [comment]
> Your step function failed. I will review the rest of the PR, but you might want to look into that whilst I do :)

Yes, same issue as the other ticket it gives OOM error at Impute stage. the data is also fine as I deleted the branch data just in case and copied new data.

**Review by MarionJHolloway — APPROVED** [review]
